### PR TITLE
Fix flaky test: hold CQ until all workloads are observed

### LINKED ISSUE #1543: Flaky: Scheduler when Scheduling workloads on clusterQueues Should admit workloads according to their priorities
**What happened**:

Flaky test `Scheduler when Scheduling workloads on clusterQueues Should admit workloads according to their priorities`

**What you expected to happen**:

Test to pass

**How to reproduce it (as minimally and precisely as possible)**:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1541/pull-kueue-test-integration-main/1742606522124341248

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind bug
/kind flake

#### What this PR does / why we need it:

Add a hold in the CQ, observe all workloads in metrics and then remove the hold.

This is more reliable than the older approach of creating the localQueue after the workloads, because we can't guarantee Kueue observes workloads and queue updates in order.

#### Which issue(s) this PR fixes:

Fixes #1543

#### Special notes for your reviewer:

While this issue has always existed, it's not possible to backport, because the stopPolicy is a new feature.
Nevertheless, the failure rate of the test is extremely low.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by tenzen-y** [comment]
This improvement seems to produce other flaky tests...

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1544/pull-kueue-test-integration-main/1742650644793135104

**Comment by alculquicondor** [comment]
Uhm.. probably not flaky but 100% failing, which is better :)
Let me take a look.

File: test/integration/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -372,44 +379,55 @@ var _ = ginkgo.Describe("Scheduler", func() {
...
 
-		ginkgo.It("Should admit workloads according to their priorities", func() {
-			queue := testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(prodClusterQ.Name).Obj()
+		ginkgo.When("Hold at startup", func() {
+			ginkgo.BeforeEach(func() {
+				cqsStopPolicy = ptr.To(kueue.Hold)
+			})
+			ginkgo.AfterEach(func() {
+				cqsStopPolicy = nil
+			})
```
Comment (alculquicondor) [inline test/integration/scheduler/scheduler_test.go:388]: I needed to reset this :)

**Review by tenzen-y — COMMENTED** [review]
LGTM!
/lgtm
/approve

**Pull request merged**